src/proposer/proposer.cpp

```diff
@@ -29,15 +30,75 @@
 
 namespace proposer {
 
-class ProposerStub : public Proposer {
+static std::shared_ptr<const CBlock> GenerateBlock(staking::ActiveChain &active_chain,
+                                                   staking::TransactionPicker &transaction_picker,
+                                                   proposer::BlockBuilder &block_builder,
+                                                   proposer::Logic &logic,
+                                                   CWallet &wallet,
+                                                   const staking::CoinSet &coins,
+                                                   const boost::optional<CScript> &coinbase_script) {
+
+  esperanza::WalletExtension &wallet_ext = wallet.GetWalletExtension();
+  const auto wallet_name = wallet.GetName();
```

Comment:
This should not be auto.
Except for usual complains that "type of wallet_name" is unclear here, we also have a small issue:
`GetName` returns reference to string. But `auto` defaults to value, so you make a very implicit copy of `wallet_name`.